### ISSUE TITLE
Invalid authentication information (no `strategy` set)

### ISSUE THREAD

**Issue by SeaDog505** [feathersjs/feathers#2061]
I upgraded my feathers authentication. I changed authentication.js file to this:

```
const { AuthenticationService, JWTStrategy } = require('@feathersjs/authentication');
const { LocalStrategy } = require('@feathersjs/authentication-local');
const { expressOauth } = require('@feathersjs/authentication-oauth');

module.exports = app => {
  const authentication = new AuthenticationService(app);

  authentication.register('jwt', new JWTStrategy());
  authentication.register('local', new LocalStrategy());

  app.use('/authentication', authentication);
  app.configure(expressOauth());
};
```
I added to default.json file this authentication field:

```
"authentication": {
    "entity": "user",
    "service": "users",
    "secret": "dEY3WyLzTUzpc1AHXjpVrj/la4c=",
    "authStrategies": [
      "jwt",
      "local"
    ],
    "jwtOptions": {
      "header": {
        "typ": "access"
      },
      "audience": "https://yourdomain.com",
      "issuer": "feathers",
      "algorithm": "HS256",
      "expiresIn": "1d"
    },
    "local": {
      "usernameField": "username",
      "passwordField": "password"
    },
    "cookie": {
        "enabled": true,
        "name": "feathers-jwt",
        "httpOnly": true,
        "secure": false
      }
```

My middleware it looks likie below:
```
app.use('/', cookieParser(), authenticate('jwt'), async (req, res, next) => {
		
		const { user } = req;
		console.log(user)
		try {
			
			await app.service('users').get(user._id);
			
			res.render('home', main['pl']);
			
		} catch(e){
			
			res.redirect('/login');
			
		}
	    
	});
```

I receive below error when I logged to my app.
`Invalid authentication information (no `strategy` set)`

what am i doing wrong?

**Comment by rvaitkus23** [comment]
Any luck? I get the same with new project (typescript)

**Comment by SeaDog505** [comment]
app.use('*', cookieParser(), (req, res, next) => {
		  
		  var cookies = req.cookies;
		  var token = cookies['feathers-jwt'];
		  
		  req.authentication = {
			      strategy: 'jwt',
			      accessToken: token
			    }
		  
		  next();
		  
	});

it works for me

**Comment by rvaitkus23** [comment]
Yes I have found out it hard way :)
I have used Feathers.js some time ago and passing authentication has changed since when. 
instead of passing `authorization: Bearer ${token}` it has changed to 
```
authentication: {
        accessToken: token,
        strategy: 'jwt',
      },
```
From documentation it is very unclear how to pass token in context. Only after careful read of change log I got the idea.

**Comment by SeaDog505** [comment]
can you show me your code after resolve this problem?

**Comment by daffl** [comment]
This has now been documented in https://docs.feathersjs.com/cookbook/express/view-engine.html

**Comment by daton89** [comment]
how do I solve that error? 
Invalid authentication information (no strategy set)

I don't really understand how the linked pages documents how to configure the strategy...

**Comment by daffl** [comment]
Your error is probably due to something different. Please create a new issue with a full example to reproduce.

**Comment by strarsis** [comment]
I had the same issue with recent FeathersJS app and react-admin Feathers provider (`ra-data-feathers`).

Adding `jwtStrategy: 'jwt'` or `jwtStrategy: 'local'` to the auth configuration object 
(passed to a `@feathersjs/authentication-client` instance) fixed the issue and the login worked:

`feathersClient.ts`:
```ts
import feathers from '@feathersjs/feathers';
import rest     from '@feathersjs/rest-client';
import auth     from '@feathersjs/authentication-client';

const hostURL = 'http://localhost:3030'; // for testing

const feathersApp = feathers();

const restClientF = rest(hostURL);

feathersApp.configure(restClientF.fetch(window.fetch));
feathersApp.configure(auth({
  storage:    window.localStorage, // Passing a WebStorage-compatible object to enable automatic storage on the client.
  jwtStrategy: 'jwt', // jwt|local ← required!
}));

export default feathersApp;
````
